### PR TITLE
Sphrose/asset editor multi tab

### PR DESCRIPTION
**Pull request by sphrose**
## What does this PR do?

This PR adds multi tab functionality for the Asset Editor. This code has been split from https://github.com/o3de/o3de/pull/2931 in order to separate it from the UX issues that PR now covers.

## How was this PR tested?

Manual testing.

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp [hunk 1]
```diff
@@ -306,306 +218,91 @@ namespace AzToolsFramework
...
-                    *conn = connect(
-                        this,
-                        &AssetEditorWidget::OnAssetSavedSignal,
-                        this,
-                        [conn, savedCallback]()
-                        {
-                            disconnect(*conn);
-                            savedCallback();
-                        });
+                    continue;
```
Comment (dtamkin1) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp:229]: Is this check needed or should it be followed by an else statement?

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp [hunk 2]
```diff
@@ -306,306 +218,91 @@ namespace AzToolsFramework
...
-                    }
-
-                    m_dirty = false;
-
-                    AddRecentPath(targetFilePath);
-
-                    SetStatusText(Status::assetCreated);
-
-                    return true;
+                    continue;
```
Comment (dtamkin1) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp:284]: Should these two checks also be followed by an else statement? I'm not sure I see the purpose of having them

**Review by dtamkin1 — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp [hunk 3]
```diff
@@ -306,306 +218,91 @@ namespace AzToolsFramework
...
+            }
 
-                if (!m_useDPE)
+            for (int tabIndex = m_tabs->count() - 1; tabIndex >= 0; tabIndex--)
+            {
+                AssetEditorTab* tab = qobject_cast<AssetEditorTab*>(m_tabs->widget(tabIndex));
+                if (!tab)
                 {
-                    m_propertyEditor->ForceQueuedInvalidation();
+                    continue;
```
Comment (dtamkin1) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp:279]: Same comment as above here

Comment (sphrose) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp:279]: The continue skips to the next iteration of the for loop, so no else is necessary.

**Pull request merged**